File: LAB QUIZZES/CSE221_Quiz02/_218/_218_Testing.java
import java.util.*;
import java.io.*;

// Compile and run: javac _218_Solution.java _218_Testing.java && java _218_Testing
public class _218_Testing {
    public static int N = 0, M = 0;
    public static ArrayList<Integer> list1 = new ArrayList<>(), list2 = new ArrayList<>(), res;

    public static ArrayList<Integer> solve(ArrayList<Integer> list1, ArrayList<Integer> list2) {
        int N = list1.size(), M = list2.size();
        ArrayList<Integer> res = new ArrayList<>();
        // Place your code here
        return res;
    }

    public static void main(String args[]) throws Exception {
        // // file input
        // // BufferedReader fin = new BufferedReader(new
        // // FileReader("TimeLimitExceeded.txt"));
        // // BufferedReader fin = new BufferedReader(new FileReader("CustomTest.txt"));
        // BufferedReader fin = new BufferedReader(new FileReader("WrongAnswer.txt"));
        // for (int i = 1; i <= 4; ++i)
        // fin.readLine();
        // fin.skip(9);
        // for (String temp : fin.readLine().replace("]", "").split(", "))
        // list1.add(Integer.parseInt(temp));
        // fin.skip(9);
        // for (String temp : fin.readLine().replace("]", "").split(", "))
        // list2.add(Integer.parseInt(temp));
        // fin.close();

        // stdin
        // 5 6
        // 1 2 2 3 4
        // 2 2 3 3 5 6
        BufferedReader fin = new BufferedReader(new InputStreamReader(System.in));
        String[] tempA = fin.readLine().split(" ");
        N = Integer.parseInt(tempA[0]);
        M = Integer.parseInt(tempA[1]);
        for (String temp : fin.readLine().split(" "))
            list1.add(Integer.parseInt(temp));
        for (String temp : fin.readLine().split(" "))
            list2.add(Integer.parseInt(temp));
        fin.close();

        // // hardcoded input
        // list1 = new ArrayList<>(Arrays.asList(1, 2, 2, 3, 4));
        // list2 = new ArrayList<>(Arrays.asList(2, 2, 3, 3, 5, 6));

        long start = System.nanoTime();
        res = solve(new ArrayList<>(list1), new ArrayList<>(list1));
        long finish = System.nanoTime();
        long elapsed = finish - start;

        // // file output
        // BufferedWriter fout = new BufferedWriter(new
        // FileWriter("OutputOfYourCode.txt"));
        // fout.write(res.toString());
        // fout.close();

        // stdout
        System.out.println(res);

        System.out.printf("Execution time for the targeted one test is %.9fs\n", elapsed * 1e-9);
        System.exit(0);
    }
}

File: LAB QUIZZES/CSE221_Quiz02/_218/_218_Testing.py
import os, sys, ast, time, random, threading, subprocess
# Compile and run: pypy _218_Testing.py
# from _218_Solution import solve
N = 0; M = 0
list1 = []; list2 = []; res = []
def solve(list1, list2):
    N = len(list1)
    M = len(list2)
    k = N + M
    res = [None]*k
    idx = k - 1

    i = 0
    j = 0
    while i < N and j < M:
        if list1[i] <= list2[j]:
            res[idx] = list1[i]
            idx -= 1
            i += 1

        else:
            res[idx] = list2[j]
            idx -= 1
            j += 1
        if idx + 2 < k:

            if res[idx+1] == res[idx+2]:
                res[idx+1] = 0
                idx += 1

    

    while i < N:
        res[idx] = list1[i]
        i += 1
        idx -= 1
        if res[idx+1] == res[idx+2]:
                res[idx+1] = 0
                idx += 1
    while j < M:
        res[idx] = list2[j]
        j += 1
        idx -= 1
        if res[idx+1] == res[idx+2]:
                res[idx+1] = 0
                idx += 1
    res = res[idx+1::]

    
    
    return res
    
    
    return res
    return res

# # file input
# # with open("TimeLimitExceeded.txt") as fin:
# # with open("CustomTest.txt") as fin:
# with open("WrongAnswer.txt") as fin:
#     for i in range(4):
#         fin.readline()
#     list1 = ast.literal_eval(fin.readline()[8:])
#     list2 = ast.literal_eval(fin.readline()[8:])

# stdin
# 5 6
# 1 2 2 3 4
# 2 2 3 3 5 6
N, M = map(int, "5 6".split())
list1 = list(map(int, "1 2 2 3 4".split()))
list2 = list(map(int, "2 2 3 3 5 6".split()))

# # hardcoded input
# list1 = [1, 2, 2, 3, 4]
# list2 = [2, 2, 3, 3, 5, 6]

start = time.time()
res = solve(list1.copy(), list2.copy())
finish = time.time()
elapsed = finish - start

# # file output
# with open("OutputOfYourCode.txt", "w") as fout:
#     fout.write(str(res))

# stdout
print(res)

print("Execution time for the targeted one test is " + f"{elapsed:.9f}" + "s", flush=True)
os._exit(0)